src/DotNetLightning.Core/Payment/PaymentRequest.fs

```diff
@@ -79,12 +79,12 @@ module private Helpers =
     let prefixValues = prefixes |> Map.toList |> List.map(fun (_, v) -> v)
         
     let checkAndGetPrefixFromHrp (hrp: string) =
-        let maybePrefix = prefixValues |> List.filter(fun p -> hrp.StartsWith(p)) |> List.tryExactlyOne
+        let maybePrefix = prefixValues |> List.filter(fun p -> hrp.StartsWith(p))
```

Comment:
You should add polyfill for `tryExactlyOne` in Utils/Extensions.fs` , in that way the change will be much simpler.

Reply:
or use System.Linq's `.Single()`